server/src/main/java/org/elasticsearch/index/mapper/Mapper.java

```diff
@@ -147,4 +147,9 @@ public final String simpleName() {
     /** Return the merge of {@code mergeWith} into this.
      *  Both {@code this} and {@code mergeWith} will be left unmodified. */
     public abstract Mapper merge(Mapper mergeWith);
+
+    /**
+     * Returns the largest column position assigned from itself and its children(if any).
+     */
+    public abstract Integer maxPosition();
```

Comment:
Could this return a `int` instead of `Integer`, or in which cases do we need a `null` result?

Reply:
When the object mapper is `ArrayMapper` or `ObjectArrayMapper`.
Arrays store the position values within `inner`.

Reply:
Can't we return that value then?